pkg/ui/query.go

```diff
@@ -36,20 +38,35 @@ type thanosVersion struct {
 	GoVersion string `json:"goVersion"`
 }
 
-func NewQueryUI(logger log.Logger, flagsMap map[string]string) *Query {
+func NewQueryUI(logger log.Logger, storeSet *query.StoreSet, flagsMap map[string]string) *Query {
 	cwd, err := os.Getwd()
 	if err != nil {
 		cwd = "<error retrieving current working directory>"
 	}
 	return &Query{
-		BaseUI:   NewBaseUI(logger, "query_menu.html", template.FuncMap{}),
+		BaseUI:   NewBaseUI(logger, "query_menu.html", queryTmplFuncs()),
+		storeSet: storeSet,
 		flagsMap: flagsMap,
 		cwd:      cwd,
 		birth:    time.Now(),
 		now:      model.Now,
 	}
 }
 
+func queryTmplFuncs() template.FuncMap {
+	return template.FuncMap{
+		"since": func(t time.Time) time.Duration {
+			return time.Since(t) / time.Millisecond * time.Millisecond
+		},
+		"formatTimestamp": func(timestamp int64) string {
+			if timestamp >= math.MaxInt64 || timestamp == -1 {
```

Comment:
hm.. `-1` timestamps is techincally a valid timestamp

Reply:
It was to handle the minTime, maxTime that could be `-1`. I guess we could deal with a pointer here

Reply:
Also, `9223372036854775807` is also a valid timestamp :D